### PR TITLE
server: fix handshake timeout logic

### PR DESCRIPTION
**Pull request by aqrln**
Clear handshake timeout on successful handshake and start sending
heartbeat messages on successful handshake handler instead of handshake
timeout handler.

Fixes: https://github.com/metarhia/jstp/issues/217
Fixes: https://github.com/metarhia/jstp/issues/283

### CODE REVIEW

File: test/node/regress-gh-217.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  applications: [app],
+  heartbeatInterval: 100,
+});
+
+tap.plan(1);
+
+server.listen(() => {
+  jstp.net.connect(
+    'app', null, server.address().port,
+    (error, connection) => {
```
Comment (nechaido) [inline test/node/regress-gh-217.js]: Why don't you place it on previous line?

File: test/node/regress-gh-283.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  applications: [app],
+  heartbeatInterval: 100,
+});
+
+tap.plan(2);
+
+server.listen(() => {
+  jstp.net.connect(
+    'app', null, server.address().port,
+    (error, connection) => {
```
Comment (nechaido) [inline test/node/regress-gh-283.js]: DItto.

File: test/node/regress-gh-217.js (hunk 1 above)
Comment (aqrln) [inline test/node/regress-gh-217.js]: I find it more readable like this ¯\_(ツ)_/¯

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/jstp/commit/e1efb911436af364be6d6c5d849ca6e0b2b7f401.

**Pull request closed without merging**